### PR TITLE
Pr wind est work queue

### PR DESCRIPTION
**Pull request by RomanBapst**
This is a replacement of #9041 and implements the standalone wind and airspeed scale factor estimator as a work queue. Previously, the estimator was living in the sensors module which did not really make sense.

Advantages: 
- the estimator is now a separate module and can be run only for VTOL and fixed wing configs
- we can extend this module to take data of multiple airspeed sensors and run multiple estimators and then use this information for airspeed failure detection

Disadvantages:
- we have some overhead in terms of memory due to the work queue and the fact that we need to subscribe to three topics, one of them is local_position which is rather large

@dagar @priseborough FYI

### CODE REVIEW

File: src/modules/wind_estimator/wind_estimator_main.cpp [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+	orb_unadvertise(_wind_est_pub);
+}
+
+int
+WindEstimatorModule::task_spawn(int argc, char *argv[])
+{
+	/* schedule a cycle to start things */
+	work_queue(LPWORK, &_work, (worker_t)&WindEstimatorModule::cycle_trampoline, nullptr, 0);
+
+	// wait until task is up & running (the mode_* commands depend on it)
```
Comment (bkueng) [inline src/modules/wind_estimator/wind_estimator_main.cpp]: `(the mode_* commands depend on it)` can be removed

File: src/modules/wind_estimator/wind_estimator_main.cpp [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+	static WindEstimatorModule *instantiate(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int custom_command(int argc, char *argv[]);
+
+	/** @see ModuleBase */
+	static int print_usage(const char *reason = nullptr);
+
+	/**
+	 * run the main loop: if running as task, continuously iterate, otherwise execute only one single cycle
```
Comment (bkueng) [inline src/modules/wind_estimator/wind_estimator_main.cpp]: This comment is incorrect as well

**Comment by dagar** [comment]
> Disadvantages:
> 
> we have some overhead in terms of memory due to the work queue and the fact that we need to subscribe to three topics, one of them is local_position which is rather large

The work queue should be a slight improvement. Previously in sensors you were keeping copies of local_position, attitude, airspeed in the sensors object, and updating the subscriptions constantly (sensors task usually running at 250 Hz). Now you only pay for it during each iteration of the wind estimator at a significantly lower rate.

**Comment by RomanBapst** [comment]
@PX4/testflights Could you please give this a test flight, preferably when it's windy.
Thanks!

**Comment by RomanBapst** [comment]
@PX4/testflights I forgot to mention that you can't test it on fmu-v2 due to lacking flash space.

**Comment by r0gelion** [comment]
@RomanBapst our FW has a fmu-v3 (Pixhawk mini). It's a rainy day, but as soon as the weather allows it we will test it.

Regards.

**Comment by dagar** [comment]
Merge the ECL side and update this PR?

**Comment by r0gelion** [comment]
@RomanBapst @dagar Is this ready for flight testing?

**Comment by RomanBapst** [comment]
@r0gelion It's good that you ask, no this is not ready. I just came back from a test flight and there seems to be a problem with the work queue which i need to discuss with @dagar 
I'll ping you as soon as we solve it.

**Comment by dagar** [comment]
The issue here was mixing different work queues for the initial and repeated cycles. See https://github.com/PX4/Firmware/pull/9122

**Comment by RomanBapst** [comment]
@dagar Thanks! I just tested it and it works fine now. Below are the results of the wind estimate using only sideslip fusion (no airspeed data).

These were the forecasted wind conditions at the time of the test flight:
wind: NE direction @ 10 knots (5.1 m/s)
windy_zurich

Luckily there was a wind sock near the flying field which confirmed the wind coming out of NE.
The big building to the left of the wind sock lies roughly in north direction.

whatsapp image 2018-03-21 at 11 09 41 am

Results of the estimator when flying circles:

Wind component estimate and absolute value:


wind_estimator_beta



Sideslip innovation & innovation standard deviation
wind_sideslip_innov

Ground speed and vehicle heading
wind_groundspeed

The absolute value estimate of 5 m/s seems quite accurate and also the wind components suggest wind out of Northeast, North-northeast.
Two separate test flight were conducted and the results of both were similar:

https://logs.px4.io/plot_app?log=a842cef4-03f8-4722-92db-6bf38f7d7788
https://logs.px4.io/plot_app?log=46e98042-87c9-4117-9f8f-fe4536ff7ca6

**Comment by dagar** [comment]
For reference here's what it looks like running in the low priority work queue (first log).

image

Which means that the cycle is being called every 101317us on average (~ 10Hz), and average runtime for each cycle is 159 microseconds. The rate skews a bit because the end of every cycle after the filter update schedules another 100 ms later.

**Comment by dagar** [comment]
Update ecl to current master and this should be good to go.

**Pull request merged**